lib/membrane/rtp/jitter_buffer.ex

```diff
@@ -194,8 +194,12 @@ defmodule Membrane.RTP.JitterBuffer do
     %{timestamp: rtp_timestamp} = buffer.metadata.rtp
     timestamp_base = state.timestamp_base || rtp_timestamp
 
+    # timestamps in RTP don't have to be monotonic therefore there can be
+    # a situation where in 2 consecutive packets the latter smaller timestamp
+    # then the previous packet while not overflowing timestamp number
+    # https://datatracker.ietf.org/doc/html/rfc3550#section-5.1
     timestamp_base =
-      if rtp_timestamp < state.previous_timestamp,
+      if rtp_timestamp - state.previous_timestamp < -@max_timestamp / 2,
```

Comment:
I suppose we have another case to consider here - we can receive a packet from before the overflow. Let's assume `rtp_timestamp = @max_timestamp - 10, state.previous_timestamp = 10`. In this case, we'll generate a huge gap in timestamps, while received a packet from ~20 ticks before

Reply:
Right, I think we got a big problem here. Without additional information we can't know if the consecutive packets are in the right order and if their timestamps are correct, when it happens inside the range it is easy to detect, but for when we approach overflow border I guess we can't do much about it...